lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -1474,8 +1474,11 @@ void LocalEnforcer::update_charging_credits(
         session->receive_charging_credit(credit_update_resp, &uc);
     session->set_tgpp_context(credit_update_resp.tgpp_ctx(), &uc);
 
+    // Here we will decide if a traffic should be activated/un-suspended
     bool should_activate = session->is_credit_ready_to_be_activated(credit_key);
-    if (!should_activate) {
+    auto typeOfCredityValidity =
+        ChargingGrant::is_valid_credit_response(credit_update_resp);
+    if (!should_activate || typeOfCredityValidity == TRANSIENT_ERROR) {
```

Comment:
Should we just put the TRANSIENT ERROR check inside of `is_credit_ready_to_be_activated` ?

Reply:
I was thinking on that but then we have to pass the response to the `is_credit_ready_to_be_activated`

My reasoning is, `is_credit_ready_to_be_activated ` is calculated out of a  current state of the session. `credit_validity_type == TRANSIENT_ERROR` is not a session state. 

So `credit_validity_type == TRANSIENT_ERROR` is just a condition that invalidates any state derived from the session 

That is why I didn't wanted to mix it and i wanted to make it explicit in the condition